clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -242,16 +196,18 @@ export const hasSavedTcfPreferences = (experience: PrivacyExperience) =>
 export const shouldResurfaceConsent = (
   experience: PrivacyExperience,
   cookie: FidesCookie
-) => {
-  if (
-    experience.component === ComponentType.TCF_OVERLAY &&
-    experience.meta?.version_hash
-  ) {
-    return experience.meta.version_hash !== cookie.tcf_version_hash;
+): boolean => {
+  if (experience.component === ComponentType.TCF_OVERLAY) {
+    if (experience.meta?.version_hash) {
+      return experience.meta.version_hash !== cookie.tcf_version_hash;
+    }
+    // Ensure we always resurface consent for TCF if for some reason experience does not have version_hash
+    return true;
   }
+  // If not every notice has previous user consent, we need to resurface consent
   return Boolean(
-    experience?.privacy_notices?.some(
-      (notice) => notice.current_preference == null
+    !experience.privacy_notices?.every((notice) =>
+      noticeHasConsentInCookie(notice, cookie)
     )
```

Comment:
I think we need to be a little careful here for these cases:
1. `experience.privacy_notices = undefined` returns true when it probably shouldn't (`!undefined = true`)
2. `experience.privacy_notices = []` returns true when it probably shouldn't (`every` on an empty array returns true)